subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c

```diff
@@ -114,55 +91,59 @@ static int cert_provision(void)
 	if (cert_size > KB(4)) {
 		LOG_ERR("Certificate too large");
 		__ASSERT(false, "Certificate too large");
-		goto exit;
+		return -ENFILE;
 	}
 
 	ret = modem_key_mgmt_exists(CONFIG_NRF_PROVISIONING_ROOT_CA_SEC_TAG,
-					MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN, &exists);
+				    MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN, &exists);
 	if (ret) {
 		__ASSERT_NO_MSG(false);
 		LOG_ERR("Failed to check for certificates err %d", ret);
-		goto exit;
+		return ret;
 	}
 
 	/* Don't overwrite certificate if one has been provisioned */
-	if (exists || !cert_size) {
-		ret = 0;
-		goto exit;
+	if (exists) {
+		LOG_DBG("Certificate already provisioned");
+		return 0;
 	}
 
-	prev_mode = mm.cb(LTE_LC_FUNC_MODE_OFFLINE, mm.user_data);
+	if (!cert_size) {
+		LOG_ERR("Provided certificate size is 0");
+		return -EINVAL;
+	}
 
-	if (prev_mode < 0) {
-		LOG_ERR("Can't put modem to offline modem for writing certificate");
-		ret = prev_mode;
-		goto exit;
+	ret = nrf_provisioning_notify_event_and_wait_for_modem_state(
+		120, NRF_PROVISIONING_EVENT_NEED_LTE_DEACTIVATED, callback_local);
```

Comment:
Maybe it's better to use macro instead of magic number 120. There are few other places where the number is used.

Reply:
Added a kconfig